addons/pos_mercado_pago/static/src/app/payment_mercado_pago.js

```diff
@@ -0,0 +1,219 @@
+        const RETRY_DELAY = 1000; // Delay between retries in milliseconds for the "ON_TERMINAL" BUG
+        try {
+            // During payment creation, user can't cancel the payment intent
+            line.set_payment_status("waitingCapture");
+
+            // Call Mercado Pago to create a payment intent
+            const payment_intent = await this.create_payment_intent();
+
+            // In case of error, don't update 'this.payment_intent' and handle the error
+            if (!("id" in payment_intent)) {
+                switch(payment_intent.status) {
+                    case 409:
+                        // We don't return on this error case, we continue but
+                        // without updating 'this.payment_intent'
+                        this._showMsg("Payment already sent to terminal, check it...", 'information');
+                        break;
+                    case 400: // Payment must be >= 500 [units]
+                    case 401: // Invalid token 
+                    default:
+                        this._showMsg(payment_intent.message, 'error');
+                        return false;
+                  }
+            } else {
+                this.payment_intent = payment_intent
+            }
+
+            // After payment creation, make the payment intent canceling possible
+            line.set_payment_status("waitingCard");
+
+            var val;
+            do {
+                // Payment intend is on the terminal, wait for transaction status change.
+                // + set a timeout in case of no notification
+                // + add a special case when it's no more possible to cancel a payment from the frontend
+                //   but only from the terminal (user has pushed on the "Cobrar" blue button)
+                val = await Promise.any([
+                    new Promise(resolve => {setTimeout(() => resolve("timeout"), 60000);}),
+                    new Promise(resolve => {this.webhook_resolver = resolve;}),
+                    new Promise(resolve => {this.payment_cancel_error_resolver = resolve;}),
+                    new Promise(resolve => {this.payment_cancel_ok_resolver = resolve;}),
+                ]);
+                // Check result
+                switch(val) {
+                    case "timeout":
+                        throw `There was a connection problem. PLEASE CANCEL PAYMENT ON THE TERMINAL 
+                               otherwise it will not be taken into account on the Odoo system.`;
+
+                    case "mp_msg_received":
+                        // We received a notification that Mercado Pago sent a webhook message
+                        // to the backend, get the last status
+                        // If no id in this.payment_intent that means user reload the page
+                        // or it is an old webhook -> trash
+                        if ("id" in this.payment_intent) {
```

Comment:
useless condition because you test it at line 70

Reply:
To be discussed